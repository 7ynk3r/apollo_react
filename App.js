import React from 'react';
import {
  ApolloClient,
  ApolloProvider,
  createNetworkInterface,
} from 'react-apollo';
import Hello from './Hello'

const uri = 'https://zvn57lnl7.lp.gql.zone/graphql';
const networkInterface = createNetworkInterface({ uri });
const client = new ApolloClient({ networkInterface });
const App = () => (
  <ApolloProvider client={client}>
    <Hello />
  </ApolloProvider>
);

export default App;
