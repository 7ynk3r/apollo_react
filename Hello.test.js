import 'react-native';
import React from 'react';
import App from './App';

const sleep = (time) => new Promise(sucess => setTimeout(sucess, time));

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('app renders correctly', async () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
  await sleep(2000)
  expect(tree).toMatchSnapshot();
});
