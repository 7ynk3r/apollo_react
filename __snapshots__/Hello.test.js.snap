// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    {
  "variables": {},
  "loading": true,
  "networkStatus": 1
}
  </Text>
</View>
`;

exports[`app renders correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    {
  "variables": {},
  "loading": false,
  "networkStatus": 7,
  "hello": "Hello world!"
}
  </Text>
</View>
`;
